Guide-‐-Event-Manager.md

```diff
@@ -15,37 +15,45 @@ However, you probably want to do something in this event. Let's cover some of th
-- "ease" - Used to interpolate values. Useful for score incrementing or movement. Also see ref_table, ref_value and ease_to for how to set this.
-- "before" - Will run immediately, but if it doesn't complete before the set amount of time, it will be cancelled. Also see delay
+- `"immediate"` (default) - Runs as soon as possible
+- `"after"` - Runs after a set amount of time specified in `delay`
+- `"before"` - Runs immediately. The event takes at least the amount of time specified in `delay` to finish
+- `"condition"` - Waits until the condition is met (namely, when `config.ref_table[config.ref_value] == config.stop_val`). You probably want to use `immediate` instead.
+- `"ease"` - Used to interpolate values. Useful for score incrementing or movement. Acts something like `config.ref_table[config.ref_value] = config.func(ease_function(original_value, config.ease_to, config.duration))`
 
 blocking - boolean: Whether or not this event may block other events. Default is true.
 
 blockable - boolean: Whether or not this event may be blocked by other events. Default is true.
 
-func - function: The function to call for the event. When this is called depends on the trigger type.
-    This function takes no arguments. 
-    It returns whether it completed. If true, then the event is over, if false, then it will be called again next time the event manager processes events (the next frame).
+func - function: The body of the event. This is where you will perform any actions or custom delays. Once the event starts, this will be run once per frame until it returns true. This can be omitted without consequence.
     Its behavior for each trigger is as follows:
-- immediate - Called when events are next processed
-- after - Called when the time is over
-- condition - Behaves like immediate. Providing a function will overwrite the default condition behaviour. If you want to do stuff conditionally and use a function, then just do your check and return false if the condition is not satisfied.
-- ease - Called each frame with the current value of the ease. The function defintion is a bit different here. The first argument is the current value of the ease. The return value is then set to the value stored in the table. The default function just returns the current value.
-- "before" - Called immediately. If the event does not complete after the delay, then it is automatically cancelled.
+- `immediate` - Called when the event begins
+- `after` - Called when the time is over
+- `condition` - Behaves exactly like immediate. Providing a function will overwrite the default condition behaviour.
+- `ease` - Called each frame with the current interpolated value so you can modify the easing function. The default function returns its argument unmodified.
+- `before` - Called when the event begins
```

Comment:
The information removed here is something I would consider important.

Reply:
The information removed is untrue. As far as I can tell, the game waits for both the time and the function, whichever takes longer.

Reply:
That's correct. I re-reviewed the code and I believe I missread the check for it. I don't quite like this description of it though, but it's consistent with the others so I'll leave it

Reply:
I've now clarified the description of `before` a bit in the `trigger` section (where IMO the difference belongs).